### PR TITLE
Ar 196 delete records

### PR DESCRIPTION
**Pull request by davidedmonds**
With this PR, we can handle record deletions. However, we have also removed a lot of the existing functionality that was present in the plugin. With this branch merged, we will only be able handle events with AVRO Record Keys and Values, and will no longer be creating an envelope over on DynamoDB. We've also removed the Kafka attributes for the moment.

### CODE REVIEW

File: src/main/java/dynamok/sink/AmazonDynamoDBFactory.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.slf4j.LoggerFactory;
+
+class AmazonDynamoDBFactory {
+    private static final Logger log = LoggerFactory.getLogger(AmazonDynamoDBFactory.class);
+
+    static AmazonDynamoDB build(ConnectorConfig connectorConfig) {
+        AmazonDynamoDBClientBuilder builder = AmazonDynamoDBClientBuilder.standard()
+                .withRequestHandlers(new PrintRequestHandler2());
+
+        if (connectorConfig.accessKeyId.value().isEmpty() || connectorConfig.secretKey.value().isEmpty()) {
```
Comment (gkalogiros-nutmeg) [inline src/main/java/dynamok/sink/AmazonDynamoDBFactory.java]: @davidedmonds I can see you use the `getPassword` method of Kafka's `AbstractConfig` to resolve the access key id. Not sure what the default behavior of Connect is but does the `isEmpty` method returns false for both the following cases?
1. config not present
2. config value empty

Comment (davidedmonds) [reply src/main/java/dynamok/sink/AmazonDynamoDBFactory.java]: Yes, in this case, as both have default values of `""`. If the default value was non-empty though, this behaviour wouldn't hold though.

File: pom.xml [hunk 2]
```diff
@@ -98,6 +98,11 @@
             <artifactId>commons-lang3</artifactId>
             <version>${apache-commons-lang3.version}</version>
         </dependency>
+        <dependency>
```
Comment (adrianmcmichael-nutmeg) [inline pom.xml]: Out of curiosity what other than the map building in your tests needs Guava?

File: src/main/java/dynamok/sink/DynamoDbClient.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                        .map(list -> list.stream().collect(Collectors.groupingBy(this::tableName,
+                                Collectors.mapping(this::toWriteRequest, Collectors.toList()))))
+                        .map(dynamoDB::batchWriteItem)
+                        .filter(result -> !result.getUnprocessedItems().isEmpty())
+                        .map(BatchWriteItemResult::getUnprocessedItems)
+                        .reduce((acc, next) -> {
+                            acc.putAll(next);
+                            return acc;
+                        })
+                        .map(UnprocessedItemsException::new);
```
Comment (adrianmcmichael-nutmeg) [inline src/main/java/dynamok/sink/DynamoDbClient.java]: Would it make it more useful for whoever catches this to know the keys of rthe records that were unprocessed as a part of the exception?

File: pom.xml (hunk 2 above)
Comment (davidedmonds) [reply pom.xml]: Looks like nothing - I seem to remember adding it to the main classpath at one point. I'll move it back to test scope.

File: src/main/java/dynamok/sink/DynamoDbClient.java (hunk 3 above)
Comment (davidedmonds) [reply src/main/java/dynamok/sink/DynamoDbClient.java]: Yep, almost certainly - have gone for putting the entire `WriteRequest` that fails in logs, which contains the full object sent over (all fields for a Put, key only for a Delete) as trying to manipulate the payload of the `WriteRequest` to extract just the key needs a reasonably deep understanding of the schema.

Comment (adrianmcmichael-nutmeg) [reply src/main/java/dynamok/sink/DynamoDbClient.java]: so trying to follow it through the commit, I see that when this is caught it logs the number of retries left then I can see that when goes to 0 it wraps and throws a ConnectionException that wraps this one. Is there a logger higher up that then logs this then?

File: src/main/java/dynamok/sink/DynamoDbClient.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        } else {
+            dynamoDB.putItem(tableName, toMap(sinkRecord));
+        }
+    }
+
+    void writeBatch(Collection<SinkRecord> records) throws UnprocessedItemsException {
+        records.forEach(this::validate);
+        Optional<UnprocessedItemsException> unprocessedItems =
+                Lists.partition(new ArrayList<>(records), config.batchSize).stream()
+                        .map(list -> list.stream().collect(Collectors.groupingBy(this::tableName,
```
Comment (javierholguera) [inline src/main/java/dynamok/sink/DynamoDbClient.java]: Can you make this one a separate method to improve a bit the readability of this Stream piece? It's a bit challenging at the moment.

File: src/main/java/dynamok/sink/DynamoDbClient.java (hunk 3 above)
Comment (javierholguera) [inline src/main/java/dynamok/sink/DynamoDbClient.java]: I think moving this line into the `if` below can be more readable? So the output of the Stream is the list of things that didn't go through and then check if the exception has to be thrown based on unprocessed records list content

File: src/main/java/dynamok/sink/DynamoDbClient.java (hunk 4 above)
Comment (davidedmonds) [reply src/main/java/dynamok/sink/DynamoDbClient.java]: Done - agreed that its a nested stream op, and is pretty wordy :(

File: src/main/java/dynamok/sink/DynamoDbClient.java (hunk 3 above)
Comment (davidedmonds) [reply src/main/java/dynamok/sink/DynamoDbClient.java]: Done

Comment (davidedmonds) [reply src/main/java/dynamok/sink/DynamoDbClient.java]: Yes - presently, a `ConnectionException` will cause connect to "fail" the connector, which logs the exception that caused the failure in the UI and (presumably, though I've not checked) the Kafka Connect logs.

Have cleaned things up further as using the Map's `toString` produces almost the same output, for much less code.

**Pull request merged**